### DISCUSSION TITLE
Official support for Laravel Sail?

### DISCUSSION THREAD

**Discussion by ryanshipswell** [laravel-zero/laravel-zero#356] [Ideas]
Any chance of "official" support for Laravel Sail? I manually added it in as a dev dependency and can confirm it works for me. I can even run my cli app with it using `./vendor/bin/sail up -d` then `./vendor/bin/sail php <app-name> <command> [args]`.

I would be willing to write some docs on my process, which basically boils down to

1. pull down a local copy of this repo with git
2. Add dev dependency using `docker run --rm  -u "$(id -u):$(id -g)" -v $(pwd):/var/www/html -w /var/www/html laravelsail/php80-composer:latest composer require laravel/sail --dev --ignore-platform-reqs` (modified from: https://laravel.com/docs/8.x/sail#installing-composer-dependencies-for-existing-projects)
3. install composer dependancies using `docker run --rm -u "$(id -u):$(id -g)" -v $(pwd):/var/www/html -w /var/www/html laravelsail/php80-composer:latest composer install --ignore-platform-reqs` (obtained from: https://laravel.com/docs/8.x/sail#installing-composer-dependencies-for-existing-projects)
4. Manually write a basic docker-dompose.yml file for the container in lieu of using artisan sail:install. (I used the basic config documented for Statamic at: https://statamic.dev/installing/docker)

Now that container will contain some unnecessary stuff for a cli tool, for example it contains npm. Potentially a modified Dockerfile could be made that would not install such unnecessary components in the container.

**Comment by owenvoke** [comment]
@x7ryan, would you want to add this as a new Component for Laravel Zero? 🤔
